google-compute-service.js

```diff
@@ -186,6 +180,32 @@ module.exports = class GoogleComputeService {
     }
   }
 
+  async getAddressResourceByIP(searchIP, region, project) {
+    const addressesClient = new compute.AddressesClient({ credentials: this.credentials });
+
+    try {
+      const request = removeUndefinedAndEmpty({
+        filter: `address="${searchIP}"`,
+        project: project || this.projectId,
+        region,
+      });
+
+      const iterable = addressesClient.listAsync(request);
+
+      try {
+        for await (const item of iterable) {
+          if (item.address === searchIP) { return item; }
+        }
+      } catch (error) {
+        return Promise.reject(error);
+      }
+
+      return undefined;
```

Comment:
If you want this function to return the promise, you need to be consistent. You should never return undefined in the first place, but this functino is pretending to be a promise, so if you want it finish here, please use Promise.resolve(), otherwise it makes no sense.